src/libsyncengine/jobs/network/upload_session/uploadsessiondrive.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Infomaniak kDrive - Desktop
+ * Copyright (C) 2023-2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "uploadsessiondrive.h"
+#include "utility/utility.h"
+
+namespace KDC {
+
+UploadSessionDrive::UploadSessionDrive(int driveDbId, std::shared_ptr<SyncDb> syncDb, const SyncPath &filepath,
+                                       const SyncName &filename, const NodeId &remoteParentDirId, SyncTime modtime,
+                                       bool liteSyncActivated, uint64_t nbParalleleThread /*= 1*/)
+    : AbstractUploadSession(filepath, filename, nbParalleleThread),
+      _remoteParentDirId(remoteParentDirId),
+      _liteSyncActivated(liteSyncActivated),
+      _modtimeIn(modtime),
+      _syncDb(syncDb),
+      _driveDbId(driveDbId) {
+    _uploadSessionType = UploadSessionType::Standard;
+}
+
+UploadSessionDrive::~UploadSessionDrive() {
+    if (_vfsForceStatus) {
+        if (!_vfsForceStatus(getFilePath(), false, 100, true)) {
```

Comment:
You may combine the two `if` statements in a single one, using `&&`.

Reply:
[fc7a0fb](https://github.com/Infomaniak/desktop-kDrive/pull/139/commits/fc7a0fbe2a75aea07978ab0a3e5783915ada9ece)